scripts/portraits/batch_extract_portraits.py

```
@@ -0,0 +1,210 @@
+    blue_bars = detect_blue_bars(screenshot, BLOOD_BAR_ROI)
+
+    if len(blue_bars) != 4:
+        print(f"  警告: 检测到 {len(blue_bars)} 个血条，预期4个")
+        return False
+
+    # 验证血条位置是否符合公差数列
+    x_coords = [bar['center_x'] for bar in blue_bars]
+    diffs = [x_coords[i+1] - x_coords[i] for i in range(len(x_coords)-1)]
+
+    if not all(abs(d - diffs[0]) < 0.005 for d in diffs):
+        print(f"  警告: 血条x坐标不符合等差数列")
```

Ruff (0.16.2)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)